src/frequenz/sdk/timeseries/_periodic_feature_extractor.py

```diff
@@ -0,0 +1,414 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""
+A class that extracts features out of historical data.
+
+It creates a profile from periodically occurring windows in a buffer of
+historical data.
+
+The profile is created out of all windows that are fully contained in the
+underlying buffer, e.g. the moving window, with the same start and end time
+modulo a fixed period.
+
+Example:
+    Consider a timeseries $T$ of historical data and sub-series $S_i$ of $T$
+    all having the same size $l$ and the same fixed distance $p$ called
+    period, where period of two sub-windows is defined as the distance of two
+    points at the same position within the sub-windows.
+
+    This class calculates a statistical profile $S$ over all $S_i$, i.e. the
+    value of $S$ at position $i$ is calculated by performing a certain
+    calculation, e.g. an average, over all values of $S_i$ at position $i$.
+"""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+import numpy as np
+from numpy.typing import NDArray
+
+from frequenz.sdk._internal._math import is_close_to_zero
+
+from ._moving_window import MovingWindow
+from ._ringbuffer import OrderedRingBuffer
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class RelativePositions:
+    """
+    A dataclass that holds the relative positions of a window in a buffer.
```

Comment:
Same here, maybe:

```suggestion
    The relative positions of a window in a buffer.
```

BTW, positions of what? By reading the class doc for me it is not clear what this is actually for.

Reply:
I've reworked the documentation by adding more context.